test/ApiCheck.Test/ApiListingComparerTests.cs

```diff
@@ -80,6 +79,68 @@ public void Compare_Detects_TypeGenericAritybreakingChanges_as_removal()
             Assert.Contains(expected, breakingChanges);
         }
 
+        [Theory]
+        [InlineData("public class ComparisonScenarios.ClassToRemoveFieldsFrom")]
+        [InlineData("public struct ComparisonScenarios.StructToRemoveFieldsFrom")]
+        public void Compare_DetectsAllFieldRemovals(string typeToCheck)
+        {
+            // Arrange
+            var v1ApiListing = CreateApiListingDocument(V1Assembly);
+            var v2ApiListing = CreateApiListingDocument(V2Assembly);
+            var comparer = new ApiListingComparer(v1ApiListing, v2ApiListing);
+
+            // Oops. The NewInternalProperty addition is a breaking change; makes it impossible to subclass type in
+            // another assembly.
+            var expected = new List<BreakingChange>
+            {
+                // Changing a const's value doesn't cause a binary incompatibility but often causes problems.
+                new BreakingChange(
+                    typeToCheck,
+                    "public const System.Int32 ConstToChangeValue = 1",
+                    ChangeKind.Removal),
+                // Removing a const doesn't cause a binary incompatibilty but often causes problems.
+                new BreakingChange(
+                    typeToCheck,
+                    "public const System.Int32 ConstToMakeField = 2",
+                    ChangeKind.Removal),
+                // Oops. Making a field writable is not technically a breaking change.
```

Comment:
And the oops here is that API Check is disallowing a non-breaking change to a field. This one is also a corner case and it's also easy to work around (with entries in the breaking change files).